src/olympia/devhub/tests/test_views.py

```diff
@@ -895,16 +902,21 @@ def test_my_addons(self):
             assert addon_item.length == 1
             assert addon_item.find('.addon-name').attr('href') == (
                 self.addon.get_dev_url('edit'))
-            if self.addon.is_listed:
+            if self.addon.is_listed and self.addon.current_version:
                 # We don't display a link to the inexistent public page for
                 # unlisted addons.
                 assert addon_item.find('p').eq(3).find('a').attr('href') == (
                     self.addon.current_version.get_url_path())
-            if self.addon.is_listed:
-                assert 'Queue Position: 1 of 1' == (
-                    addon_item.find('p').eq(4).text())
-            assert addon_item.find('.upload-new-version a').attr('href') == (
-                reverse('devhub.submit.version', args=[self.addon.slug]))
+            if not self.addon.is_disabled:
+                if self.addon.is_listed:
+                    assert 'Queue Position: 1 of 1' == (
+                        addon_item.find('p').eq(4).text())
+
+                assert addon_item.find(
+                    '.upload-new-version a').attr('href') == (
+                    reverse('devhub.submit.version', args=[self.addon.slug]))
+            else:
+                assert not addon_item.find('.upload-new-version a')
```

Comment:
This test method has waaaaaay to manny `if` statements and lots of potential to skip things that should fail loud. Can we refactor this a bit to make this simpler to reason about, less prone to fail silently?